tests/test_incrementapprover.py

```diff
@@ -186,12 +186,36 @@ def test_approval_if_there_are_only_ok_openqa_jobs(caplog: LogCaptureFixture, mo
 
 
 @responses.activate
-@pytest.mark.usefixtures("fake_ok_jobs", "fake_product_repo")
 def test_skipping_if_rescheduling(caplog: LogCaptureFixture, monkeypatch: MonkeyPatch) -> None:
-    run_approver(caplog, monkeypatch, reschedule=True)
-    last_message = [x[-1] for x in caplog.record_tuples][-1]
-    assert "have passed" not in last_message
-    assert "Re-scheduling jobs for" in last_message
+    # Mock the openQA API to return existing failed jobs.
+    # These would normally block approval, but reschedule=True should ignore them.
+    resp = responses.add(
+        GET,
+        openqa_url,
+        json={"done": {"passed": {"job_ids": [20]}, "failed": {"job_ids": [21]}}},
+    )
```

Comment:
This should use `fake_not_ok_jobs` instead of repeating that code here.

Reply:
We shouldn't accept this PR as long as it's introducing this duplication. How you solve it is up to your preferences